app/models/search/query.rb

```diff
@@ -32,6 +32,76 @@ def count
     )['count']
   end
 
+  # Retrieve a randomly sampled selection of results:
+  def sample(count: 5)
+    response = $elasticsearch.search(
+      index: index_name,
+      type: document_type,
+      body: {
+        query: {
+          function_score: {
+            query: filtered_query,
+            random_score: {},
+            boost_mode: "replace"
+          }
+        },
+        size: count
+      }
+    )
+
+    QueryResult.new(klass, response, { page: 1, per_page: count })
+  end
+
+  # Perform a specific aggregation:
+  def aggregation_search(aggregation)
+    $elasticsearch.search(
+      index: index_name,
+      type: document_type,
+      body: {
```

Comment:
Not sure if we want this before or after ES7. I did a cursory check anyway and I don't think there's any breaking changes here between 6 and 7.

There's a [minor optimization](https://www.elastic.co/guide/en/elasticsearch/reference/7.x/search-aggregations-bucket-composite-aggregation.html#_early_termination) that might be worth a TODO for ES7:
> In order to optimize the early termination it is advised to set `track_total_hits` in the request to false. The number of total hits that match the request can be retrieved on the first request and it would be costly to compute this number on every page.

Reply:
It seems like that optimization only applies when there's an index on the field being summarized (otherwise early termination is impossible).  Are you suggesting adding an index on `fandom_ids`?

Reply:
Ah gotcha, that optimization has a more specific use case than I thought. The current setup is fine.